lib/src/utils/parsing/parsing_utils.dart

```diff
@@ -0,0 +1,321 @@
+    : parseCoin(tx.treasury, InvalidDataReason.treasuryNotValid);
+
+  // donation
+  final donation = tx.donation == null
+    ? null
+    : parseUint64Str(tx.donation, {'min': '1', 'max': maxLovelaceSupplyStr}, InvalidDataReason.donationNotValid);
+
+  return ParsedTransaction(
+    network: network,
+    inputs: inputs,
+    outputs: outputs,
+    ttl: ttl == null ? null : BigInt.parse(ttl),
+    auxiliaryData: auxiliaryData,
+    validityIntervalStart: validityIntervalStart == null ? null : BigInt.parse(validityIntervalStart),
+    withdrawals: withdrawals,
+    certificates: certificates,
+    fee: BigInt.parse(fee),
+    mint: mint,
+    scriptDataHashHex: scriptDataHashHex,
+    collateralInputs: collateralInputs,
+    requiredSigners: requiredSigners,
+    includeNetworkId: includeNetworkId,
+    collateralOutput: collateralOutput,
+    totalCollateral: totalCollateral == null ? null : BigInt.parse(totalCollateral),
+    referenceInputs: referenceInputs,
+    votingProcedures: votingProcedures,
+    treasury: treasury == null ? null : BigInt.parse(treasury),
+    donation: donation == null ? null : BigInt.parse(donation),
+  );
+}
+
+ParsedNetwork parseNetwork(Network network) {
+  final parsed = ParsedNetwork(
+    protocolMagic: parseUint32t(
+      network.protocolMagic,
+      InvalidDataReason.networkInvalidProtocolMagic.message,
+    ),
+    networkId: parseUint8t(
+      network.networkId,
+      InvalidDataReason.networkInvalidNetworkId.message,
+    ),
+  );
+  validate(
+    parsed.networkId <= 0x0F,
+    InvalidDataReason.networkInvalidNetworkId.message,
+  );
+  return parsed;
+}
+
+int parseUint32t(dynamic value, String errMsg) {
+  validate(isUint32(value), errMsg);
+  return value as int;
+}
+
+
+bool isUint32(dynamic data) =>
+  isInteger(data) && data >= 0 && data <= 4294967295;
+
+bool isInteger(dynamic data) =>
+  data is int || data is BigInt;
+
+bool isUint8(dynamic data) =>
+  isInteger(data) && data >= 0 && data <= 255;
+
+int parseUint8t(dynamic value, String errMsg) {
+  validate(isUint8(value), errMsg);
+  return value as int;
+}
+
+String parseCoin(dynamic coin, InvalidDataReason errMsg) {
+  return parseUint64Str(coin, {'max': maxLovelaceSupplyStr}, errMsg);
+}
+
+
+String parseUint64Str(dynamic val, Map<String, String> constraints, InvalidDataReason errMsg) {
+  if (val is String) {
+    validate(isUint64Str(val) && isUintStr(val, constraints), errMsg.message);
+    return val;
+  } else if (val is int) {
+    validate(isUint64Number(val) && isUintStr(val.toString(), constraints), errMsg.message);
+    return val.toString();
+  } else if (val is BigInt) {
+    validate(isUint64Bigint(val) && isUintStr(val.toString(), constraints), errMsg.message);
+    return val.toString();
+  } else {
+    validate(false, errMsg.message);
+    return ''; // This line is unreachable but required for Dart analysis.
+  }
+}
+
+bool isUint64Str(dynamic data) =>
+  isUintStr(data, {});
+
+bool isUint64Number(dynamic data) =>
```

Comment:
for int and BigInt you should not have any `parse` functions at all. You should only have functions checking that the size is within boundaries.

So you only need:

```dart
// it **should** not be allowed to call this method for `int` 
//   because the type of data shouldn't be int if you expect it 
//   to be 64 bits. It means you needed BigInt to begin with
bool isUint64(BigInt data) {...}
bool isInt64(BigInt data) {...}

// You prob don't need a BigInt alternative for those. Cause you should have used `int` if you don't expect the size to be above 32 bits
bool isUint32(int data) {...}
bool isInt32(int data) {...}
bool isUint16(int data) {...}
bool isInt16(int data) {...}
bool isUint8(int data) {...}
bool isInt8(int data) {...}
```

* You should not need a `parse` that takes a number and returns String. 
* You should not need to use BigInt.parse or BigInt.from anywhere in the SDK code (except for defining constants). 
* You should not need `int.parse` or `int.tryParse` anywhere (except for ipStringToBytes)